src/nni_manager/training_service/kubernetes/azureStorageClientUtils.ts

```diff
@@ -131,67 +146,82 @@ export namespace AzureStorageClientUtility {
                 }
-            } catch(error) {
+            } catch (error) {
                 deferred.reject(error);
+
                 return deferred.promise;
             }
         }
         // All files/directories are copied successfully, resolve
         deferred.resolve();
+
         return deferred.promise;
     }
-    
+
     /**
      * downlod a directory from azure
-     * @param fileServerClient 
-     * @param azureDirectory 
-     * @param azureShare 
-     * @param localDirectory 
+     * @param fileServerClient
+     * @param azureDirectory
+     * @param azureShare
+     * @param localDirectory
      */
-    export async function downloadDirectory(fileServerClient: any, azureDirectory:any, azureShare: any, localDirectory: any): Promise<void>{
+    export async function downloadDirectory(fileServerClient: any, azureDirectory: string, azureShare: any, localDirectory: string):
+     Promise<void> {
         const deferred: Deferred<void> = new Deferred<void>();
-        mkDirP(localDirectory);
-        fileServerClient.listFilesAndDirectoriesSegmented(azureShare, azureDirectory, 'null', function(error: any, result: any, response: any) {
-            if(('entries' in result) === false){
-                getLogger().error(`list files failed, can't get entries in result`);
+        await mkDirP(localDirectory);
+        fileServerClient.listFilesAndDirectoriesSegmented(azureShare, azureDirectory, 'null',
+                                                          async (error: any, result: any, response: any) => {
+            if (('entries' in result) === false) {
+                getLogger()
+                  .error(`list files failed, can't get entries in result`);
                 throw new Error(`list files failed, can't get entries in result`);
             }
 
-            if(('files' in result['entries']) === false){
-                getLogger().error(`list files failed, can't get files in result['entries']`);
+            if (('files' in result.entries) === false) {
+                getLogger()
+                  .error(`list files failed, can't get files in result['entries']`);
                 throw new Error(`list files failed, can't get files in result['entries']`);
             }
 
-            if(('directories' in result['directories']) === false){
-                getLogger().error(`list files failed, can't get directories in result['entries']`);
+            if (('directories' in result.directories) === false) {
+                getLogger()
+                  .error(`list files failed, can't get directories in result['entries']`);
                 throw new Error(`list files failed, can't get directories in result['entries']`);
             }
 
-            for(var fileName of result['entries']['files']){
+            // tslint:disable: no-floating-promises
+            for (const fileName of result.entries.files) {
```

Comment:
There are some floating promises problems here, may cause bugs.  Please do not disable  floating promises warnings.